collector/src/main/java/com/bitmovin/analytics/stateMachines/PlayerStateMachine.java

```diff
@@ -27,6 +28,7 @@
     private Handler heartbeatHandler = new Handler();
     private int heartbeatDelay = 59700; // default to 60 seconds
     private final BitmovinAnalytics analytics;
+    private VideoStartFailedReason videoStartFailedReason;
```

Comment:
can we move this away from the State Machine?
Because the state machine does not intentionally work with the values of the state
for me this is more of a thing for the adapter

Reply:
We have multiple possible reasons why play attempt failed (timeout, some error...), and this reason is determined in some of the listeners in`BitmovinSdkAdapter`. We need some way of passing that information to the `BitmovinAnalytics`.

One option was to use similar logic as it now in case of onError event (`ErrorCode class`), but then this class would need some changes, and I'm not sure if it makes sense to use it for this case.

I'm open for any suggestions/improvements for this case.